promise-types/json/json_promise_type.py

```diff
@@ -0,0 +1,152 @@
+
+        model = self.create_attribute_object(attributes)
+        if (
+            model.object
+            and isinstance(model.object, str)
+            and not is_json_serializable(model.object)
+        ):
+            raise ValidationError(
+                "'{}' is not a valid data container".format(model.object)
+            )
+
+        if model.array:
+            if isinstance(model.array, str):
+                if not is_json_serializable(model.array):
+                    raise ValidationError(
+                        "'{}' is not a valid list".format(model.array)
+                    )
+
+                if not isinstance(json.loads(model.array), list):
+                    raise ValidationError(
+                        "'{}' is not a valid data".format(model.array)
+                    )
+
+            elif not isinstance(model.array, list):
+                raise ValidationError(
+                    "'{}' is not a valid data array".format(model.array)
+                )
+
+        if model.number and not is_number(model.number):
+            raise ValidationError(
+                "'{}' is not a valid int or real".format(model.number)
+            )
+
+        if model.primitive and model.primitive not in ["true", "false", "null"]:
+            raise ValidationError(
+                "expected 'true', 'false' or 'null' but got '{}".format(model.primitive)
+            )
+
+    def evaluate_promise(self, promiser, attributes, metadata):
+        model = self.create_attribute_object(attributes)
+        filename, _, field = promiser.partition(":")
+
+        # type conversion
+
+        datatype = next(t for t in self.types if t in attributes)
+
+        match datatype:
+            case "object" | "array":
+                data = (
+                    json.loads(attributes[datatype])
+                    if isinstance(attributes[datatype], str)
+                    else attributes[datatype]
+                )
+            case "number":
+                data = float(model.number) if "." in model.number else int(model.number)
+            case "primitive":
+                data = None if model.primitive == "null" else model.primitive == "true"
+            case _:  # strings
+                data = attributes[datatype]
+
+        # json manipulation
+
+        try:
+            with open(filename, "r+") as f:
+                content = json.load(f)
+        except (FileNotFoundError, json.JSONDecodeError):
+            content = {}
+
+        if field:
+            if field in content and content[field] == data:
+                return (Result.KEPT, ["correct_json"])
+            content[field] = data
+        else:
+            if content == data:
+                return (Result.KEPT, ["correct_json"])
+            content = data
+
+        with open(filename, "w") as f:
+            json.dump(content, f, indent=4)
+
+        self.log_info("Updated '{}'".format(filename))
+        return (Result.REPAIRED, ["written_json"])
```

Comment:
As mentioned in Slack - I don't think it's necessary to use the result classes feature in your module, it doesn't provide any additional information. If users want to define classes based on the outcome, they can use a classes body.

```suggestion
        return Result.REPAIRED
```